src/payment/strategies/googlepay/googlepay-checkoutcom-initializer.ts

```diff
@@ -0,0 +1,144 @@
+        paymentMethod: PaymentMethod,
+        hasShippingAddress: boolean
+    ): Promise<GooglePayPaymentDataRequestV2> {
+        this.publishableKey = paymentMethod.initializationData.checkoutcomkey;
+        this.testMode = !!paymentMethod.config.testMode;
+
+        return this._mapGooglePayCheckoutcomDataRequestToGooglePayDataRequestV2(
+            checkout,
+            paymentMethod.initializationData,
+            hasShippingAddress
+        );
+    }
+
+    teardown(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    async parseResponse(paymentData: GooglePaymentData): Promise<TokenizePayload> {
+        let token;
+        try {
+            token = JSON.parse(paymentData.paymentMethodData.tokenizationData.token);
+        } catch (err) {
+            throw new InvalidArgumentError('Unable to parse response from Google Pay.');
+        }
+
+        if (!token.signature || !token.protocolVersion || !token.signedMessage) {
+            throw new PaymentMethodFailedError('Unable to parse response from Google Pay.');
+        }
+        const finalToken = await this._convertToken(this.testMode, this.publishableKey, token);
+
+        return Promise.resolve({
+            nonce: finalToken,
+            type: 'CreditCard',
+            description: paymentData.paymentMethodData.description,
+            details: {
+                cardType: paymentData.paymentMethodData.info.cardNetwork,
+                lastFour: paymentData.paymentMethodData.info.cardDetails,
+            },
+        });
+    }
+
+    private _convertToken(testMode: boolean, checkoutcomkey: string, token: CheckoutcomGooglePayToken): Promise<string> {
+        if (!token || !token.protocolVersion) {
+            throw new PaymentMethodFailedError('Unable to parse response from GooglePay.');
+        }
+
+        return this._requestCheckoutcomTokenize(testMode, checkoutcomkey, {
+            type: 'googlepay',
+            token_data: token,
+        }).then(checkoutcomToken => {
+            if (!checkoutcomToken || !checkoutcomToken.token) {
+                throw new PaymentMethodFailedError('Unable to parse response from Checkout.com');
+            }
+
+            return checkoutcomToken.token;
+        });
+    }
+
+    private _requestCheckoutcomTokenize(testMode: boolean, checkoutcomKey: string, data = {}): Promise<CheckoutcomToken> {
+        const TEST_URL: string = 'https://api.sandbox.checkout.com/tokens';
+        const LIVE_URL: string = 'https://api.checkout.com/tokens';
+
+        const url = testMode ? TEST_URL : LIVE_URL;
+
+        const requestSender = createRequestSender();
+        const response = requestSender.post(url, {
+            credentials: false,
+            body: data,
+            headers: {
+                'Content-Type': ContentType.Json,
+                Authorization: checkoutcomKey,
+                'X-XSRF-TOKEN': null,
+            },
+        });
+
+        return response.then(res => {
```

Comment:
Please use `async/await` in this file. We don't want a new file to have a mixture of `async/await` vs `then`.

Reply:
Changed to async/await